### PR TITLE
feat: implement  x-tagGroup feature

### PR DESCRIPTION
**Pull request by bencagri**
## Description
Docusaurus should support nested  sidebar which already implemented in [ReDocly](https://redocly.com/docs/api-reference-docs/specification-extensions/x-tag-groups/) as x-tagGroups.

## Motivation and Context
Most of the complex Apis need nested resources in one openapi spec. This feature aims to solve this problem within one resource instead of having multiple openapi resources.

## How Has This Been Tested?

jest tests added for valid openapi files.

## Screenshots (if appropriate)

```yaml
tags:
  - name: tag1
    description: Everything about your restaurant
    x-displayName: Tag 1
  - name: tag2
    description: Tag 2 description
    x-displayName: Tag 2
  - name: tag3
    description: Tag 3 description
    x-displayName: Tag 3

x-tagGroups:
  - name: Tag 1 & 2
    tags:
      - tag1
      - tag2
  - name: Trinity
    tags:
      - tag1
      - tag2
      - tag3
  - name: Last Two
    tags:
      - tag2
      - tag3

```
Screenshot 2024-03-12 at 14 46 02

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by sserrata** [comment]
Hi @bencagri, looks like there are a few linter errors and some new merge conflicts. Let me know if I can assist with resolving these issues, thanks!

**Comment by bencagri** [comment]
hi @sserrata, conflicts were resolved and also linter errors fixed.

File: yarn.lock
Comment (sserrata) [inline yarn.lock:1]: One (hopefully) last thing: can you revert `yarn.lock` to the version on `main` and rerun the linter/formatter? Aside from introducing a few new packages I suspect the linter rules may have also changed, resulting in updating the import order on a number of files unrelated to your changes. Thanks!

**Comment by sserrata** [comment]
@bencagri, any plans/thoughts on supporting generated category index pages, similar to how non-x-tag-groups are handled?

**Comment by bencagri** [comment]
> @bencagri, any plans/thoughts on supporting generated category index pages, similar to how non-x-tag-groups are handled?

hi Steven, actually this should be document writer's choice -from my point of view-. The generator should act as proof of concept, not to generate the docs that are not given. 

if you think this is a must, then to provide such a feature, we **might** generate docs under something _uncategorized_ even if category has not been configured.

But again, this wouldn't be a part of its job. What is your thought?

**Comment by sserrata** [comment]
Makes sense to me. I was speaking more about the tag groups that are generated, i.e. whether or not you think they should link to index pages when they are clicked.

**Comment by bencagri** [comment]
hmm. i missed this point. correct, i dont think we should provide index pages, as they are already a category, no?

**Comment by sserrata** [comment]
Sorry, it took me a minute to remember the name of the option: `categoryLinkSource`

docs: https://docusaurus-openapi.tryingpan.dev/#sidebaroptions

reference: https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/blob/54399a2b85b25b9bb754558d0e3ca61b0e9bb27b/packages/docusaurus-plugin-openapi-docs/src/sidebars/index.ts#L149

**Comment by bencagri** [comment]
hi @sserrata, i think we should point that `categoryLinkSource` is disregardred  if the `groupPathsBy` is set as `tagGroup`

**Review by sserrata — APPROVED** [review]

**Comment by Muneer-Shafi** [comment]
@sserrata it shows introduction in every dropdown/tagGroup and contains same info please have a look

**Pull request merged**